application/common/models/User.php

```diff
@@ -196,44 +120,110 @@ private function validatePassword(): Closure
     {
         return function ($attributeName) {
 
-            if ( ! $this->hasPasswordAlready()) {
+            if ($this->password_id === null) {
                 $this->attemptPasswordMigration();
             }
 
-            if (! password_verify($this->password, $this->password_hash)) {
+            /** @var Password $currentPassword */
+            $currentPassword = $this->getPassword()->one() ?? new Password();
```

Comment:
You might want to add a `getCurrentPassword()` method to make sure (or maybe just make it clear) that the one you're getting is the current one, not one the possibly many of this `User`'s passwords stored in the `Password` table.